docs/user-guide/docker/building-in-docker.rst

```diff
@@ -0,0 +1,244 @@
+
+.. note::
+
+   You can see a list of all the versions and tags of the Rasa Docker image
+   `here <https://hub.docker.com/r/rasa/rasa/>`_.
+
+
+.. _model_training_docker:
+
+Training a Custom Rasa Model with Docker
+########################################
+
+Edit the ``config.yml`` file to use the pipeline you want, and place
+your NLU and Core data into the ``data/`` directory.
+Now you can train your Rasa model by running:
+
+.. code-block:: bash
+
+  docker run \
+    -v $(pwd):/app \
+    rasa/rasa:latest-full \
+    train \
+      --domain domain.yml \
+      --data data \
+      --out models
+
+Here's what's happening in that command:
+
+  - ``-v $(pwd):/app``: Mounts your project directory into the Docker
+    container so that Rasa can train a model on your training data
+  - ``rasa/rasa:latest-full``: Use the Rasa image with the tag ``latest-full``
+  - ``train``: Execute the ``rasa train`` command within the container. For more
+    information see :ref:`command-line-interface`.
+
+In this case, we've also passed values for the location of the domain file, training
+data, and the models output directory to show how these can be customized.
+You can also leave these out since we are passing the default values.
+
+.. note::
+
+    If you are using a custom NLU component or policy, you have to add the module file to your
+    Docker container. You can do this by either mounting the file or by including it in your
+    own custom image (e.g. if the custom component or policy has extra dependencies). Make sure
+    that your module is in the Python module search path by setting the
+    environment variable ``PYTHONPATH=$PYTHONPATH:<directory of your module>``.
+
+Adding Custom Actions
+*********************
+
+To create more sophisticated assistants, you will want to use :ref:`custom-actions`.
+Continuing the example from above, you might want to add an action which tells
+the user a joke to cheer them up.
+
+Running a Custom Action in Docker-Compose
+#########################################
+
+Start by creating the custom actions in a directory ``actions``:
+
+.. code-block:: bash
+
+  mkdir actions
+  # Rasa SDK expects a python module.
+  # Therefore, make sure that you have this file in the directory.
+  touch actions/__init__.py
+  touch actions/actions.py
+
+Then build a custom action using the Rasa SDK, e.g.:
+
+.. code-block:: python
+
+  import requests
+  import json
+  from rasa_sdk import Action
+
+
+  class ActionJoke(Action):
+    def name(self):
+      return "action_joke"
+
+    def run(self, dispatcher, tracker, domain):
+      request = requests.get('http://api.icndb.com/jokes/random').json()  # make an api call
+      joke = request['value']['joke']  # extract a joke from returned json response
+      dispatcher.utter_message(text=joke)  # send the message back to the user
+      return []
+
+Next, add the custom action in your stories and your domain file.
```

Comment:
Instead of describing this in sentences below, can we just show what it would look like?

Reply:
Do you think we need to show what the new data/stories.md looks like? otherwise its covered